### ISSUE TITLE
Neovide not responding to configurations set in init.lua

### ISSUE THREAD

**Issue by Ktoks** [neovide/neovide#2620]
**Describe the bug**
Neovide isn't responding to configuration changes upon reload in an init.lua file.

**To Reproduce**
1. Make any changes to configuration.
2. Start Neovide.

**Expected behavior**
The configuration should be noticeably different with many configurations.

**Screenshots**
![Screenshot from 2024-05-31 17-35-42](https://github.com/neovide/neovide/assets/36412717/63a88be2-3337-4fe6-8953-0863c3857159)

**Desktop:**
 - OS - Pop!_OS 22.04 
 - Kernel - 6.8.0-76060800daily20240311-generic
 - Neovide Version - 0.13.1
 - Neovim Version - v0.10.1-dev-34+g58e1ef2f6

**Please run `neovide --log` and paste the contents of the `.log` file created in the current directory here:**
[neovide_rCURRENT.log](https://github.com/user-attachments/files/15519484/neovide_rCURRENT.log)

**Additional context**
The image above both shows the problem - transparency when my Opacity is set to 1.0, and it shows my init.lua file.

**Comment by fredizzimo** [comment]
I think you forgot to tell how you are starting Neovim, because it does not seem to be started by Neovide, due to the lack of the line `Starting neovim with` in the logs. Neovide also gets assigned channel id 3 instead of 1, suggesting that something else is already attached.

This probably means that your `init.lua` has already been run when Neovide starts and at that point `vim.g.neovide` won't be set, so none of your options will apply.

You can either remove the `if vim.g.neovide` check, or if you you are starting with `--listen`, make sure that you don't add the `--headless` option, since then it won't wait for the UI to attach before loading the configuration.

**Comment by Ktoks** [comment]
`neovide --log` is the exact command I used just now... New log as a result from that command(same result).
[neovide_rCURRENT.log](https://github.com/user-attachments/files/15523114/neovide_rCURRENT.log)

**Comment by fredizzimo** [comment]
This time the log shows that the settings were changed 11 time, so I believe that everything else than the transparency applied.

Window transparency and full screen are incompatible due the optimizitions the compositor does, so there's nothing to show below the window, I recommend maximizing the window instead.

**Comment by Ktoks** [comment]
You are right, I do see the other settings are now working, the transparency is still not fixed even when I set full screen to false and put `--maximized` in the command line.
[neovide_rCURRENT.log](https://github.com/user-attachments/files/15523227/neovide_rCURRENT.log)

**Comment by fredizzimo** [comment]
The `neovide_transparency` variable is a bit poorly named https://neovide.dev/configuration.html#transparency, it should actually be `neovide_opacity', so with your value of `1.0` for `neovide_transparency` the window will be fully opaque.

**Comment by Ktoks** [comment]
> The `neovide_transparency` variable is a bit poorly named https://neovide.dev/configuration.html#transparency, it should actually be `neovide_opacity', so with your value of `1.0`for`neovide_transparency` the window will be fully opaque.

Same result with opacity instead of transparency. I've even tried using Vim script instead of Lua

**Comment by fredizzimo** [comment]
I mean the value should be less than one for a transparent window, and 0 for a completely transparent one.

**Comment by fredizzimo** [comment]
@Ktoks, did you try if `neovide_transparency=0` makes a difference? I don't see why it shouldn't. But your current setting `neovide_transparency=0` should not, since it's already the default.

**Comment by fredizzimo** [comment]
I created this Neovim issue 
* https://github.com/neovide/neovide/issues/2620

The best workaround for now is to remove the `if vim.g.neovide` check, if you use `neovide --server`

**Comment by mikew** [comment]
It's not a great workaround, but for my setup I have some auto detection of UIs, and support an `NVIM_FORCE_UI` env arg, which I used when starting the remote server for these cases:

https://github.com/mikew/vimrc/blob/e43305db7a91e4ff13f09ec40b69301b1983eb98/src/nvim/lua/vimrc.lua#L25-L48

**Comment by Ktoks** [comment]
I no longer use Neovim in this fashion.

Feel free to open your own issue if this is still occurring.

**Comment by mikew** [comment]
I'm wondering if Neovide's approach to settings-via-neovim is even feasible? Does it track changes to `vim.g`? Like these settings are intended for the client side of things, while the neovim server is a server, with multiple clients, of which one can be Neovide.
